next.config.js

```diff
@@ -4,7 +4,7 @@ import redirects from './redirects.js'
 
 const domain = process.env.SERVER_DOMAIN || process.env.VERCEL_URL
 const url = domain
-  ? `https://${domain}`
+  ? `${domain.includes('localhost') ? 'http' : 'https'}://${domain}`
```

Comment:
If incoming `domain` is coming in with `localhost`, do we need the `|| 'http://localhost:3000'` at the end of this expression?